.changeset/metal-bulldogs-destroy.md

```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/shared': patch
+---
+
+Introduce client-side caching to TelemetryCollector
```

Comment:
```suggestion
Internal change to add client-side caching to Clerk's telemetry. This prevents event flooding in frequently executed code paths, such as for React hooks or components. Gracefully falls back to old behavior if e.g. `localStorage` is not available. As such, no public API changes and user changes required.
```

You can follow [these guidelines](https://www.notion.so/clerkdev/Code-collaboration-4d32380fb78c4226abd9e1bcb7d55abf?pvs=4#62d500ed4eea46c0b8732f8fe4db6056) to write more thorough changesets. I'm also not always following them 1:1 but they are good general guardrails.